### ISSUE TITLE
Repository organization 

### ISSUE THREAD

**Issue by ZhenXW77** [amritansh22/Data-Structures-and-Algorithms-in-cpp#597]
Lots of data structure and algorithm .cpp files are not organized in a folder. For example there are multiple math related .cpp files that can be group into a math folder. String manipulation, traversal, and miscellaneous folders can also be created.

**Comment by amritansh22** [comment]
Hello @ZhenXW77 thanks a lot for pointing this issue out. Are you willing to work on this issue ?

**Comment by ZhenXW77** [comment]
Yes, I will put all the algorithms and data structures in categorized folders and I will add couple more sorting algorithms.
